src/cratedb_fivetran_destination/schema_migration_helper.py

```
@@ -310,6 +449,33 @@ def handle_table_sync_mode_migration(self, op, schema, table):
+        Validate table is non-empty and `max(_fivetran_start) < operation_timestamp`.
+        """
+        with self.engine.connect() as conn:
+            # Synchronize previous writes.
+            conn.execute(sa.text(f'REFRESH TABLE "{schema}"."{table}";'))
+
+            # Check for emptiness.
+            cardinality = int(
+                conn.execute(sa.text(f'SELECT COUNT(*) FROM "{schema}"."{table}";')).scalar_one()
+            )
+            if cardinality == 0:
+                raise ValueError("table is empty")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)